docs/jwst/error_propagation/main.rst

```diff
@@ -0,0 +1,374 @@
+<https://jwst-pipeline.readthedocs.io/en/latest/jwst/group_scale/description.html>`_
+
+
+Stage 2 pipelines 
+-----------------
+Stage 2 pipelines perform additional instrument-level and observing-mode corrections and 
+calibrations to produce fully calibrated exposures. There are two Stage 2 pipelines: one 
+for `imaging 
+<https://jwst-pipeline.readthedocs.io/en/latest/jwst/pipeline/calwebb_image2.html>`_
+and one for `spectroscopy 
+<https://jwst-pipeline.readthedocs.io/en/latest/jwst/pipeline/calwebb_spec2.html>`_
+In these pipelines, the various steps that apply corrections and calibrations
+apply those same corrections/calibrations to all variance arrays arrays.
+
+There are two unique configuration files to be used to control these pipelines, 
+depending on whether the data are to be treated as a Time Series Observation 
+(in which case some of the steps are set to be skipped by default).
+
+
+Stage 2 Imaging Processing pipeline - calwebb_image2 & calwebb_tso-image2
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+The steps executed in this pipeline depend on whether the data are to be treated
+as TSO.  For non-TSO exposures, the steps are background image subtraction,
+assign_wcs, flat_field, photom, and resample.  For TSO exposures, the steps are
+assign_wcs, flat_field, and photom.
+
+Of the steps listed above, below are the steps that alter the ERR or VAR arrays
+of the science ramp data and output those altered arrays in the science product 
+that is passed to the next step in the pipeline.  For the steps not listed below,
+those arrays do not exist or are propagated unchanged to the next pipeline step.
+
+Flat_field
+~~~~~~~~~~
+(NOTE: get from the latest online readthedocs for this step once James updates it.)
+The SCI array from the flat-field reference file is divided into the VAR_POISSON
+and VAR_RNOISE arrays of the science data set. The variance in the flat is
+calculated in this step; the ERR array is then recalculated as these three 
+variances added in quadrature.  The newly created VAR_FLAT array and the updated 
+VAR_POISSON and VAR_RNOISE are output to the science products(s). For details of
+the flat fielding correction, see `Flatfielding
+<https://jwst-pipeline.readthedocs.io/en/latest/jwst/flatfield/index.html>`_
+
+
+Photom
+~~~~~~
+The photom step is applied for the modes: imaging and non-IFU spectroscopy,
+NIRSpec IFU, and MIRI MRS. The application of the step to these modes is very
+similar - for details see `Photom
+<https://jwst-pipeline.readthedocs.io/en/latest/jwst/photom/index.html>`_
+
+For these modes, the calibration information includes a scalar
+flux conversion constant, as well as optional arrays of wavelength and
+relative response (as a function of wavelength).  The combination of the scalar
+conversion factor and the 2-D response values is applied to the science data,
+including the SCI and ERR arrays, as well as the variance (VAR_POISSON,
```

Comment:
Is the calibration actually just applied to the ERR array, or is it first applied to the variance arrays and then the ERR is recalculated from the updated variances? Should be mathematically equivalent either way, but just curious.